src/Components/Testimonials/TestimonialsForm.js

```diff
@@ -1,48 +1,191 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { useDropzone } from 'react-dropzone';
+import { useFormik } from 'formik';
+import { TextField, Box, Button, Alert, Typography } from '@mui/material';
+import { testimonialFormService } from '../../Services/testimonialsService';
+import { listHasValues, dropzoneConfig } from '../../utils';
+import '../../Styles/TestimonialsFormStyles.css';
 import '../FormStyles.css';
 
-const TestimonialForm = () => {
-  const [initialValues, setInitialValues] = useState({
-    name: '',
-    description: '',
+const TestimonialForm = ({ id }) => {
+  const [files, setFiles] = useState([]);
+  const [image, setImage] = useState('');
+  const [imageError, setImageError] = useState(false);
+  const [apiResponse, setApiResponse] = useState({});
+  const [testimonialDescription, setTestimonialDescription] = useState('');
+  const { multipleFiles, maxFiles, validImages } = dropzoneConfig;
+
+  const getTestimonial = () => ({
+    name: 'Testimonial testings',
+    description: 'This is a mockup testimonial',
+    image: '',
   });
 
-  const handleChange = (e) => {
-    if (e.target.name === 'name') {
-      setInitialValues({ ...initialValues, name: e.target.value });
+  const handleDrop = (acceptedFiles, fileRejections) => {
+    setFiles(
+      acceptedFiles.map((file) =>
+        Object.assign(file, {
+          preview: URL.createObjectURL(file),
+        }),
+      ),
+    );
+    if (fileRejections.length === 0) {
+      const reader = new FileReader();
+
+      reader.readAsDataURL(acceptedFiles[0]);
+      reader.onload = () => {
+        const base64 = reader.result;
+
+        setImage(base64);
+      };
     }
-    if (e.target.name === 'description') {
-      setInitialValues({ ...initialValues, description: e.target.value });
+  };
+
+  const handleCKeditorChange = (e, editor) => {
+    const data = editor.getData();
+
+    setTestimonialDescription(data);
+  };
+
+  const checkIfIsEditing = () => {
```

Comment:
the name of this function is not correct and also it has three responsabilities to checkIfIsEditing, get the testimonial and set the description.
you should wrap this into two functions:
```
const isEditingMode = () => id !== null;
const updateTestimonialWithCurrentData = () => {
    const currentTestimonial = getTestimonial();
    formik.values.name = currentTestimonial.name;
    setTestimonialDescription(currentTestimonial.description)
}

useEffect(() => {
    if(isEditingMode()) updateTestimonialWithCurrentData();
},[])
```